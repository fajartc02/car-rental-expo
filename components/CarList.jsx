import { View, Text, StyleSheet, Image, Pressable } from "react-native";
import React from "react";
import { Row, Col } from "./Grid";
import { Ionicons } from "@expo/vector-icons";

const formatCurrency = (number) => {
  return new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "IDR",
  }).format(number);
};

export default function CarList({
  image,
  carName,
  passengers,
  baggage,
  price,
  style,
  onPress,
}) {
  return (
    <Pressable style={{ ...styles.card }} onPress={onPress}>
      <View>
        <Row
          alignItems={"center"}
          justifyContent={"flex-start"}
          gap={16}
          style={{ padding: 16 }}
        >
          <Col>
            <Image style={styles.img} source={image} />
          </Col>
          <Col style={styles.descContainer}>
            <Text style={styles.carName}>{carName}</Text>
            <Row>
              <Col style={styles.subContainer}>
                <Ionicons name="people-outline" size={16} color="#8A8A8A" />
                <Text style={styles.subText}>{passengers}</Text>
              </Col>
              <Col style={styles.subContainer}>
                <Ionicons name="bag-outline" size={16} color="#8A8A8A" />
                <Text style={styles.subText}>{baggage}</Text>
              </Col>
            </Row>
            <Text style={styles.price}>{formatCurrency(price)}</Text>
          </Col>
        </Row>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  card: {
    marginVertical: 4,
    borderRadius: 4,
    shadowColor: "rgba(0,0,0,0.2)",
    // android
    elevation: 0.5,
    // ios
    shadowOpacity: 1,
    shadowRadius: 4,
  },
  img: {
    width: 40,
    height: 40,
    // resizeMode: "contain",
    objectFit: "contain",
  },
  carName: {
    fontSize: 14,
  },
  descContainer: {
    gap: 10,
  },
  subContainer: {
    flexDirection: "row",
    gap: 2,
    justifyContent: "flex-start",
    alignItems: "center",
  },
  subText: {
    color: "#8A8A8A",
    fontWeight: "bold",
    marginRight: 10,
  },
  price: {
    fontSize: 14,
    color: "#5CB85F",
  },
});
