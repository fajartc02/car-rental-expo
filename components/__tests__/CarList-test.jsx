import * as React from "react";
import renderer from "react-test-renderer";

import CarList from "../CarList";

it(`GeoLocation renders correctly`, () => {
  const tree = renderer
    .create(
      <CarList
        image={{
          uri: "https://firebasestorage.googleapis.com/v0/b/km-sib-2---secondhand.appspot.com/o/orders%2F1725855113022-9d222482-af8c-44a9-ab3b-ae780f03b36b.jpeg?alt=media",
        }}
        carName={"Innova"}
        passengers={10}
        baggage={2}
        price={500000}
        onPress={() => router.navigate("(list-car)/details/" + car.id)}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
