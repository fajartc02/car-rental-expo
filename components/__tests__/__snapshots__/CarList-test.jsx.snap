// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoLocation renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 4,
      "elevation": 0.5,
      "marginVertical": 4,
      "shadowColor": "rgba(0,0,0,0.2)",
      "shadowOpacity": 1,
      "shadowRadius": 4,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-start",
          "padding": 16,
        }
      }
    >
      <View>
        <Image
          source={
            {
              "uri": "https://firebasestorage.googleapis.com/v0/b/km-sib-2---secondhand.appspot.com/o/orders%2F1725855113022-9d222482-af8c-44a9-ab3b-ae780f03b36b.jpeg?alt=media",
            }
          }
          style={
            {
              "height": 40,
              "objectFit": "contain",
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          {
            "gap": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
            }
          }
        >
          Innova
        </Text>
        <View
          style={
            {
              "alignItems": "baseline",
              "flexDirection": "row",
              "gap": 0,
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 2,
                "justifyContent": "flex-start",
              }
            }
          >
            <Text />
            <Text
              style={
                {
                  "color": "#8A8A8A",
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              10
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 2,
                "justifyContent": "flex-start",
              }
            }
          >
            <Text />
            <Text
              style={
                {
                  "color": "#8A8A8A",
                  "fontWeight": "bold",
                  "marginRight": 10,
                }
              }
            >
              2
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#5CB85F",
              "fontSize": 14,
            }
          }
        >
          IDRÂ 500,000.00
        </Text>
      </View>
    </View>
  </View>
</View>
`;
